CoreVulkanDisplay/src/au/gov/asd/tac/constellation/visual/vulkan/renderables/CVKAxesRenderable.java

```diff
@@ -673,52 +650,15 @@ private void DestroyVertexBuffer() {
             cvkVertexBuffer.Destroy();
             cvkVertexBuffer = null;
         }
-    }     
-    
+    }    
+   
     
-    // ========================> Uniform buffers <======================== \\
+    // ========================> Push constants <======================== \\
     
-    private int CreateUniformBuffers(MemoryStack stack){    
+    private void UpdatePushConstants(){
         CVKAssert(cvkSwapChain != null);
-        
-        int imageCount = cvkSwapChain.GetImageCount(); 
-        
-        vertexUniformBuffers = new ArrayList<>();     
-        for (int i = 0; i < imageCount; ++i) {   
-            CVKBuffer vertUniformBuffer = CVKBuffer.Create(cvkDevice, 
-                                                          VertexUniformBufferObject.SIZEOF,
-                                                          VK_BUFFER_USAGE_UNIFORM_BUFFER_BIT | VK_BUFFER_USAGE_TRANSFER_DST_BIT,
-                                                          VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT);
             
-            vertUniformBuffer.DEBUGNAME = String.format("CVKAxesRenderable vertexUniformBuffer %d", i);               
-            vertexUniformBuffers.add(vertUniformBuffer);            
-        }
-        
-        return UpdateUniformBuffers(stack);
-    }
-    
-    private int UpdateUniformBuffers(MemoryStack stack) {
-        CVKAssert(cvkSwapChain != null);
-        int ret = VK_SUCCESS;
-     
-        int imageCount = cvkSwapChain.GetImageCount();        
-        
-        // Staging buffer so the buffer we render with can be in the most optimised memory
-        CVKBuffer cvkStagingBuffer = CVKBuffer.Create(cvkDevice, 
-                                                      VertexUniformBufferObject.SIZEOF,
-                                                      VK_BUFFER_USAGE_TRANSFER_SRC_BIT,
-                                                      VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT | VK_MEMORY_PROPERTY_HOST_COHERENT_BIT);       
-        cvkStagingBuffer.DEBUGNAME = "CVKAxesRenderable.UpdateUniformBuffers cvkStagingBuffer";  
-        
-        // LIFTED FROM AxesRenderable.reshape(...)
-        // This is a GL viewport where the screen space origin is in the bottom left corner
-        //final int[] viewport = new int[]{0, 0, cvkSwapChain.GetWidth(), cvkSwapChain.GetHeight()};
-        
-        // In Vulkan the screen space origin is in the top left hand corner.  Note we put the origin at 0, H and 
-        // the viewport dimensions are W and -H.  The -H means we we still have a 0->H range, just running in the
-        // opposite direction to GL.
-        final int[] viewport = new int[]{0, cvkSwapChain.GetHeight(), cvkSwapChain.GetWidth(), -cvkSwapChain.GetHeight()};
-               
+        final int[] viewport = new int[]{0, cvkSwapChain.GetHeight(), cvkSwapChain.GetWidth(), -cvkSwapChain.GetHeight()};             
```

Comment:
This is something I think we should put in the swapchain.  It's never going to be anything else than this and the swapchain owns the variables that build it.

Reply:
Moved to swapchain class